src/main/java/li/cil/tis3d/util/FontRendererUtils.java

```diff
@@ -26,15 +28,42 @@
      * @param tooltip the tooltip to add the info to.
      */
     public static void addStringToTooltip(final String info, final List<Text> tooltip) {
+        final MinecraftClient mc = MinecraftClient.getInstance();
+        //if (mc != null) {
+            //final TextRenderer fontRenderer = mc.textRenderer;
+            //tooltip.addAll(fontRenderer.wrapLines(info, Constants.MAX_TOOLTIP_WIDTH).stream().map(LiteralText::new).collect(Collectors.toList()));
+        //} else {
+            tooltip.add(new LiteralText(info));
+        //}
+    }
```

Comment:
> I'd say we pass new TranslatableText(key) to this instead of the translated plain string, then we can pass that to wrapLines or add it directly to tooltip.

Then you'd be wrapping a String in a `TranslatableText`, just to unwrap and rewrap it in a `LiteralText` again, or am I seeing this wrong?

Reply:
Btw I implemented it locally using a custom lambda that creates LiteralText's directly from the wrapped lines, will now look at the dimension id stuff.

Reply:
Ah, sorry, didn't really elaborate why I made that suggestion. The main thing I like about passing in a `Text` is that it makes the method more versatile  and the signature more clear about that at the same time: when taking a `Text` it's essentially agnostic to whether it gets passed an already localized string or not. When taking a string I'd probably want to look into the method implementation to make sure whether it takes a localization key or an already localized string.

(Caveat: now that I spell it out that way, it's not clear from the signature that the `Text` put into the list is always a literal text...)

Also I kinda like the symmetry in the arguments -- `Text` in, `Text` out.

All that said, I don't have strong feelings about this, just felt using the new wrappers for user-facing text are appropriate to use here; but it's really just that, a suggestion. The main thing I was referring to with "cleaning up" that class is to remove the dead code and make this method work again -- however you see fit! :)

Reply:
> When taking a string I'd probably want to look into the method implementation to make sure whether it takes a localization key or an already localized string.

Ohh, right, I see your point now. I personally got the feeling that the various `Text` subclasses were essentially "baked" strings that should not be operated on further except style annotations etc., but that's my naive interpretation having just started looking at that namespace. Since the helper function also breaks up the string passed in, that would naturally destroy any localization key, though the function name doesn't exactly make this clear. I really wish Java had `typedef` so raw strings and localization key's could be cleanly separated.

Reply:
Right, `wrapLines` has the same issue. I guess it could be made explicit via visitor pattern; instead of taking the list to append to, take a lambda that gets passed the `LiteralText`s... starts feeling pretty FizzBuzzEnterpriseEdition tho :P